src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php

```diff
@@ -292,16 +292,20 @@ public function createAction(Request $request): Response
                     );
                 }
 
-                return $this->redirectToRoute('admin_customers_view', [
-                    'customerId' => $customerId,
-                ]);
+                if ($customerId) {
```

Comment:
How is that possible that `$customerId` is null ? 🤔  I dont think it's possible to create adresses into the BO without linking them to a customer, which would consequently assign a value to `$customerId` ?

Maybe the solution here is not to handle the "what if `$customerId` is null" but rather to fetch the `$customerId` where it is.

Something like
```
$handlerResult = $addressFormHandler->handle($addressForm);

if ($handlerResult->isSubmitted() && $handlerResult->isValid()) {
    $customerId = $addressForm->getData()['customer_id'];
```

Reply:
Hi @matks 

I think :

- If you come from the customer controller, you must be redirected to the customer page after the address creation. 
- If you have created the address from address controller you must be redirected to the address page.